src/openforms/payments/tasks.py

```diff
@@ -34,6 +48,7 @@ def update_submission_payment_status(submission_id: int):
         if is_retrying:
             raise
         return
-
-    # if this properly executed, we can schedule "instant" confirmation e-mail delivery
-    send_confirmation_email.delay(submission_id)
+    else:
+        if is_retrying:
+            submission.needs_on_completion_retry = False
+            submission.save(update_fields=["needs_on_completion_retry"])
```

Comment:
Initially I thought I had done it because I couldn't coordinate the registration task and the payment status update task to set it right :thinking: I have tried again to move it in the finalisation task and it seems to work :thinking: